user_config.c

```diff
@@ -0,0 +1,96 @@
+#include "user_config.h"
+#include <stdio.h>
+#include <string.h>
+#include "usb_app.h"
+
+typedef enum {
+	CHECK_CHAR,
+	CHECK_NUM,
+	CHECK_SEMI_COLON
+} level;
+
+static level check_level = CHECK_CHAR;
+static uint8_t set_debug_level = 0;
+static uint8_t temp_num = 0;
+static bool set_sensor_messages = false;
+
+// save_type used to save what kind of message type is being checked ('G' for debug level, 'S' for sensor message, or 'L' for config)
+static char save_type;
+
+// function takes in a message and parses it to see if max_debug_level() or allow_sensor_messages() needs to be changed
+void parse_usb_string(const char *input) {
+	// Check for if the input is not NULL, if so: continue parsing the message
+	// Else: end function early
+	if (input != NULL) {
+		for (uint8_t i = 0; input[i] != '\0'; i++) {
+			switch (check_level) {
+			// Case 2: last check for if there has been a valid message sent
+			// --> if parse_usb_string has received a semicolon
+			case CHECK_SEMI_COLON:
+				check_level = CHECK_CHAR;
+				if (input[i] == ';') {
+					switch (save_type) {
+					case 'G':
+						set_debug_level = temp_num;
+						break;
+					case 'S':
+						set_sensor_messages = temp_num;
+						break;
+					case 'L':
+						;
+						char config_msg[72];
+						char sensor_check[8];
+						uint8_t msg_length = 0;
+						strcpy(sensor_check, "are");
+						if (!allow_sensor_messages())
+							strcat(sensor_check, " not");
+						msg_length = snprintf(config_msg, 72, "Current Config: Max debug level = %d & Sensor messages %s allowed!", max_debug_level(), sensor_check);
+						config_msg[71] = '\r\n';
+						config_msg[72] = '\0';
+						usb_app_write_string(config_msg, msg_length);
```

Comment:
check the docs to see if msg_length includes the null terminator, I'm not sure if it does

Reply:
It looks like snprintf includes the null terminator as long as the length is not 0 (which it should never be in our case since something is always being written into the string).

Reply:
ok, so if snprintf includes that and it's writing it to the config_msg, you shouldn't need the line
`config_msg[72] = '\0'`

Reply:
Yeah I took that out!